oshi-core/src/main/java/oshi/software/os/linux/LinuxNetworkParams.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Oshi (https://github.com/dblock/oshi)
+ *
+ * Copyright (c) 2010 - 2017 The Oshi Project Team
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ *
+ * Contributors:
+ * https://github.com/dblock/oshi/graphs/contributors
+ */
+package oshi.software.os.linux;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import oshi.software.os.NetworkParams;
+import oshi.util.ExecutingCommand;
+import oshi.util.FileUtil;
+
+public class LinuxNetworkParams implements NetworkParams{
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(LinuxNetworkParams.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getHostName() {
+        String hn = FileUtil.getStringFromFile("/proc/sys/kernel/hostname");
```

Comment:
Prefer native calls to reading files or parsing command lines.  In this case, there is a `gethostname()` function in `libc`.